src/core/database/query.helpers.ts

```diff
@@ -103,77 +103,27 @@ export function matchRequestingUser(
   ]);
 }
 
-//DELETE service
-export const setBaseNodeLabelsAndIdDeleted = (
-  query: Query,
-  baseNodeLabels: string[]
-) => {
-  //set labels as Deleted
-  baseNodeLabels.forEach((label) => {
-    query.call(setLabelDeleted, 'node', label).with('distinct(node) as node');
-  });
-
-  //set id as deleted_id
-
-  query
-    .with('*, node.id as nodeId')
-    .set({
-      variables: {
-        // eslint-disable-next-line @typescript-eslint/naming-convention
-        'node.deleted_id': 'nodeId',
-      },
-    })
-    .removeProperties({
-      node: 'id',
-    })
-    .with('distinct(node) as node');
-};
-
-export type UniqueProperties<BaseNode extends Resource> = Partial<
-  Record<keyof BaseNode, string[]>
+export type BaseNodePropertyRelationships<BaseNode extends Resource> = Array<
+  keyof BaseNode
 >;
 
-export function setPropLabelsAndValuesDeleted<BaseNode extends Resource>(
+export function deleteProperties<BaseNode extends Resource>(
   query: Query,
-  uniqueProperties: UniqueProperties<BaseNode>
+  // relationships of properties to delete
+  // it will prefix every node label with Deleted_
+  relationships: BaseNodePropertyRelationships<BaseNode>
 ) {
-  entries(uniqueProperties).forEach(([property, labels], i) => {
-    const currentPropertyNodeAlias = `propertyNode${i}`;
-    // Match the baseNode out to the propertyNode
-    query.optionalMatch([
-      node('node'),
-      relation('out', '', property),
-      node(currentPropertyNodeAlias, 'Property'),
-    ]);
-
-    // Reset all the labels on the propertyNode with the Delete_ prefix
-    labels?.forEach((label) => {
-      query.call(setLabelDeleted, currentPropertyNodeAlias, label);
-    });
-
-    // Reset all the values on the propertyNode to deleted_value
-    query
-      .with(`*, ${currentPropertyNodeAlias}.value as propVal${i}`)
-      .set({
-        variables: {
-          [`${currentPropertyNodeAlias}.deleted_value`]: `propVal${i}`,
-        },
-      })
-      .removeProperties({
-        [currentPropertyNodeAlias]: 'value',
-      })
-      .with('*');
-  });
+  return query.raw(
+    `
+    match(node)-[propertyRel:${relationships.join(
```

Comment:
I used a JS var here because I don't believe it's possible to use a `.raw()` variable as a string literal in this way. https://jamesfer.me/cypher-query-builder/classes/query.html#raw

Reply:
Is this just not supposed to be done?